### PR TITLE
Use Relayer for gasEstimates, update UI

### LINKED ISSUE #4384: Update gas price copy
Gas price will be updated from: 

"Gas cost/price" or "Estimated gas cost" 
to
 "Transaction fee" 
throughout the UI. 

Locations: 
- [x] Withdraw funds modal 
- [x] Market creation - review step (x3) 
- [x] Market creation - review step tooltips (x3)
- [x] Initial reporting form
- [x] Disputing form 

If I've missed any area change that too :)

### LINKED ISSUE #4859: Gnosis safe relayer - UI Fees/Gas Work
#### TODO
- [x] Update middleware to support direct gasEstimate calls (via relayer) **[#4955]**
- [x] Use relevant gasEstimate values to display gas estimations in the UI (see below for locations)**[#4955]**
- [x] If relayer returns an error, or falls over, fallback on our current constants as a gas estimate strategy AND convert these values via our uniswap price feed. **[#4955]**
- [x] Transactions fee modal needs to be wired up (Use Gas Relay gasStation). Gas price needs to be set by a present (FAST/AVERAGE/SLOW) or a specific Gas Price amount. [#4891]
- [x] Use gas price set in TX Fee Modal over whatever the relayer suggests for gas price [#4891]
- [x] Copy updates [#4384] **[#4955 ]**

#### UI Locations which display Fees/Gas
      - Create a Market Review (Funds required screen) ✅
      - Trading/Confirm form ✅
      - Withdraw form ✅
      - Gas Fees Modal ✅
      - Disputing Modal ✅
      - Claim Proceeds Modal
      - Get Participation Tokens Modal ✅
      - Migrate V1 to V2 REP Modal ✅

### PR DESCRIPTION
**Pull request by andrewdamelio**
## 🚙 ⛽️⛽️⛽️   🚗
closes #4859 
closes #4384

TODO:
- https://github.com/AugurProject/augur/issues/4932
- minor updates once 0x is supported

### CODE REVIEW

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts [hunk 1]
```diff
@@ -279,7 +314,7 @@ export async function uniswapRepForEthRate(rep: BigNumber): Promise<BigNumber> {
 }
 
 export async function uniswapEthForDaiRate(wei: BigNumber): Promise<BigNumber> {
-  return new BigNumber(182);
+  return new BigNumber(148);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:317]: This looks a little weird. Why 148

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts [hunk 2]
```diff
@@ -503,6 +573,51 @@ export interface CreateNewMarketParams {
...
+        params.prices,
+        params.numTicks,
+        params.extraInfo
+      );
+    }
+    case CATEGORICAL: {
+      return universe.createCategoricalMarket_estimateGas(
+        params.endTime,
+        params.feePerCashInAttoCash,
+        '0x0000000000000000000000000000000000000000',
```
Comment (bthaile) [inline packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:601]: There us a constant NULL_ADDRESS for all zero address. 
Why use different address for other address param?

File: packages/augur-ui/src/modules/create-market/components/review.tsx [hunk 3]
```diff
@@ -327,21 +336,21 @@ export default class Review extends React.Component<
             />
           </span>
 
-          <Subheaders header="Initial liquidity" subheader={"The total of the initial batch of orders you added on the previous step."} />
-          <span>
+          { s.formattedInitialLiquidityDai.value > 0 && <Subheaders header="Initial liquidity" subheader={"The total of the initial batch of orders you added on the previous step."} /> }
+          { s.formattedInitialLiquidityDai.value > 0 && <span>
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/review.tsx]: Why have another conditional using ` s.formattedInitialLiquidityDai`

File: packages/augur-ui/src/modules/create-market/components/review.tsx [hunk 4]
```diff
@@ -327,21 +336,21 @@ export default class Review extends React.Component<
...
             />
             {Gnosis_ENABLED && ethToDaiRate && <LinearPropertyLabelTooltip
-              label={'Estimated Gas Cost'}
-              value={initialLiquidityGasInDai.formattedValue + ' DAI'}
+              label={'Estimated Transaction fee'}
+              value={initialLiquidityGasInDai + ' DAI'}
             />}
             {!Gnosis_ENABLED && <LinearPropertyLabelTooltip
-              label={'Estimated Gas Cost'}
+              label={'Estimated Transaction fee'}
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/review.tsx]: Other places use “Transaction Fee”, might as well drop “Estimated” here

File: packages/augur-ui/src/modules/create-market/components/review.tsx [hunk 5]
```diff
@@ -327,21 +336,21 @@ export default class Review extends React.Component<
...
+          { s.formattedInitialLiquidityDai.value > 0 && <Subheaders header="Initial liquidity" subheader={"The total of the initial batch of orders you added on the previous step."} /> }
+          { s.formattedInitialLiquidityDai.value > 0 && <span>
             <LinearPropertyLabel
               label={"Initial Liquidity"}
               value={s.formattedInitialLiquidityDai.formattedValue + " DAI"}
             />
             {Gnosis_ENABLED && ethToDaiRate && <LinearPropertyLabelTooltip
-              label={'Estimated Gas Cost'}
-              value={initialLiquidityGasInDai.formattedValue + ' DAI'}
+              label={'Estimated Transaction fee'}
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/review.tsx]: Other places use “Transaction Fee”, might as well drop “Estimated” here

File: packages/augur-ui/src/modules/modal/reporting.tsx [hunk 6]
```diff
@@ -204,16 +209,29 @@ export default class ModalReporting extends Component<
       numOutcomes,
       marketType,
       description: '',
-      attoRepAmount: this.state.inputtedReportingStake.inputToAttoRep,
+      attoRepAmount: estimateGas ? '100000000000000000000' : this.state.inputtedReportingStake.inputToAttoRep,
```
Comment (bthaile) [inline packages/augur-ui/src/modules/modal/reporting.tsx]: Why use value of 100000000000000000000?

File: packages/augur-ui/src/modules/modal/withdraw-form.tsx [hunk 7]
```diff
@@ -74,35 +87,63 @@ export class WithdrawForm extends Component<
...
-          ? availableEth.minus(GasCosts.dai.fullPrecision)
-          : availableEth.minus(GasCosts.rep.fullPrecision);
+      if (currency === ETH && newAmount) {
+        amountMinusGas = availableEth
+          .minus(bnNewAmount)
+          .minus(fallBackGasCosts.eth.fullPrecision);
+      } else {
+        amountMinusGas =
+          currency === DAI
+            ? availableEth.minus(fallBackGasCosts.dai.fullPrecision)
```
Comment (bthaile) [inline packages/augur-ui/src/modules/modal/withdraw-form.tsx:144]: Why subtract from available ETH balance?

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). ❤️
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/vv2h3ko**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/reporting/common.tsx [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import React, { Component, useState } from 'react';
+import React, { Component, useState, useEffect } from 'react';
```
Comment (bthaile) [inline packages/augur-ui/src/modules/reporting/common.tsx]: Is useEffect used?

File: packages/augur-ui/src/modules/reporting/common.tsx [hunk 9]
```diff
@@ -493,6 +511,13 @@ export class DisputingBondsView extends Component<
     const remaining = convertAttoValueToDisplayValue(
       createBigNumber(stakeRemaining)
     );
+
+    const gasEstimate = formatGasCostToEther(
```
Comment (bthaile) [inline packages/augur-ui/src/modules/reporting/common.tsx]: Does this need to be in the render method. It doesnt relay on any user inputs

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts (hunk 1 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:317]: we are using a mock exchange rate now (until Uniswap is ready). That was the price of ETH when i started this work. The previous mock value of 182 added some uneeded confusion to testing.

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts (hunk 2 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:601]: Will update constant 👍
I was getting contract errors using NULL_ADDRESS as the address param

File: packages/augur-ui/src/modules/create-market/components/review.tsx (hunk 3 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/create-market/components/review.tsx]: Truth mistake

File: packages/augur-ui/src/modules/create-market/components/review.tsx (hunk 4 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/create-market/components/review.tsx]: 👍

File: packages/augur-ui/src/modules/modal/reporting.tsx (hunk 6 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/modal/reporting.tsx]: Meant to use a constant. Using 1 REP as the placeholder value for the estimateGas param, didn't see any difference in terms of gas usage based on this value so anything >0 would work for this estimation request.

File: packages/augur-ui/src/modules/modal/withdraw-form.tsx (hunk 7 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/modal/withdraw-form.tsx:144]: Good eye, that's a mistake!

File: packages/augur-ui/src/modules/reporting/common.tsx (hunk 9 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/reporting/common.tsx]: Nope, I can hoist this up and pass it in as a prop

File: packages/augur-ui/src/modules/reporting/common.tsx (hunk 8 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/reporting/common.tsx]: Nope, will remove.

**Comment by bthaile** [comment]
not seeing tx fee for open reporting

Screen Shot 2019-12-02 at 11 25 15 AM


👆 this got fixed. 👍

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 🎖
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/o9ph3ro**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/modal/withdraw-form.tsx (hunk 7 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/modal/withdraw-form.tsx:144]: actually, this is fine. 

We subtract from available ETH balance b/c we want to determine how much ETH is remaining once they pay the gas costs to send DAI.

**Comment by bthaile** [comment]
transaction fee looks good for market creation. 

FYI:
getting this error trying to create market, am using correct portis login

Screen Shot 2019-12-02 at 6 41 38 PM

File: packages/augur-ui/src/modules/modal/withdraw-form.tsx (hunk 7 above)
Comment (bthaile) [reply packages/augur-ui/src/modules/modal/withdraw-form.tsx:144]: `fallBackGasCosts.dai.fullPrecision` is this actually ETH transaction fee?

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts [hunk 10]
```diff
@@ -113,6 +113,22 @@ export async function convertV1ToV2() {
   return response;
 }
 
+export async function convertV1ToV2_estimate() {
+  const { contracts } = augurSdk.get();
+  const allowance = createBigNumber(99999999999999999999).times(
```
Comment (justinbarry) [inline packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:118]: Is the allowance always constant?

**Comment by bthaile** [comment]
FYI:
and got these errors when trying to sign up to create a new wallet

Screen Shot 2019-12-02 at 6 45 20 PM

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts (hunk 10 above)
Comment (bthaile) [reply packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:118]: yea, just allow max amount

**Comment by bthaile** [comment]
created new gnosis safe and got the Invalid owner error when trying to create a market

Screen Shot 2019-12-02 at 6 52 06 PM

**Comment by bthaile** [comment]
FYI:
after refreshing the UI got `SafeMultisigTxExists...` error

Screen Shot 2019-12-02 at 6 54 02 PM

**Review by bthaile — APPROVED** [review]
transaction fees are showing up correctly.

**Pull request merged**